### ISSUE TITLE
Data enrichment fails (Canada)

### ISSUE THREAD

**Issue by pramsey** [CartoDB/cartodb#10599]
### Context

* Account pramsey.carto.com

### Steps to Reproduce

* Added smallish (30k) polygon layer, and selected a data enrichment analysis
* Polygons are in Victoria, Canada, so selected the Cantonese language option from the library

### Current Result

![screenshot_57](https://cloud.githubusercontent.com/assets/113883/20072381/f0aa2c24-a4dc-11e6-9c17-5eb4f03aad84.png)

### Browser and version
* Chrome, OSX

**Comment by talos** [comment]
@pramsey Thanks for the report.  Unfortunately, until table level functions for analysis are deployed, a 30K polygon augmentation will likely time out.  The underlying functions in the observatory-extension for doing this already exist, but the interfaces for using them haven't been built yet.

If you're curious, the observatory-extension work around this (basically finished) is here: https://github.com/CartoDB/observatory-extension/pull/199 .  The accompanying camshaft ticket is here: https://github.com/CartoDB/camshaft/issues/228

The current deployment makes one call to dataservices per row, and the overhead for that multiplies very rapidly.  Table-level functions make just one call per table, generally resulting in an order of magnitude performance improvement.

We're currently looking at between 30 seconds and 60 seconds per thousand polygons right now, depending on their complexity.  That would mean your augmentation could have taken up to thirty minutes.  Did you receive the error message immediately, or did you get the loading spinner for a while first?

I've flagged with @saleiva that we should prevent users from augmenting so many rows using the UI until we've upgraded the tech behind the scenes: https://github.com/CartoDB/cartodb/issues/9798

While for the sake of a demo it's not useful, it is possible to use table-level functions in the SQL API.  It's also possible to use the publicly documented Observatory functions in bite-size portions by using a `WHERE` against cartodb_id to send off updates in ~1000 row pieces.
